srcpkgs/ell/patches/musl-rawmemchr.patch

```diff
@@ -0,0 +1,14 @@
+--- ell/pem.c.orig	2020-11-30 09:02:25.583791998 -0500
++++ ell/pem.c	2020-11-30 09:03:23.548913496 -0500
+@@ -224,7 +224,11 @@
+ 
+ 		/* Check that each header line has a key and a colon */
+ 		while (start < end) {
++#if defined(__GLIBC__)
+ 			const char *lf = rawmemchr(start, '\n');
++#else
++			const char *lf = memchr(start, '\n', end - start);
++#endif
+ 			const char *colon = memchr(start, ':', lf - start);
+ 
+ 			if (!colon)
```

Comment:
I'd just use `memchr`, any performance gain here is insignificant. This should be upstreamed as well, let me know if you can do it.

Reply:
Note we have an existing patch for `iwd` that uses `rawmemchr` for glibc and `memchr` for musl.  I just followed that convention (I am fine with just using `memchr` unconditionally).

Reply:
Huh, I hadn't noticed that one. Feel free to remove it, they added a fallback in IWD for missing rawmemchr, but forgot to add one in ELL.

Reply:
OK, I have removed the match from iwd.

Reply:
If you want, leave a comment that next release fixes this (there's an upstream commit too, if you'd like to cherry-pick it: https://git.kernel.org/pub/scm/libs/ell/ell.git/commit/?id=3916c1cc32be205e185a8a56de5cde1b9ec60e81 )

Reply:
Ok, I've made a note in the current patch.  

I did not cherry-pick upstream's change because I believe I would also need to run autotools (autoconf) a pre_configure hook to make it go into effect (changes modify configure.ac but not configure) and don't feel it is that necessary.  Please correct me if I am wrong.

Reply:
Sounds good to me, thanks :)